### PR TITLE
exp/services/webauth: change JWT key config option to a JWK

### PR DESCRIPTION
**Pull request by leighmcculloch**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What
Change to using the JSON Web Key (JWK) format for configuring the JWK key used by the webauth service to sign JWTs.

### Why
The JSON Web Key (JWK) format is defined in [RFC7517] and makes it easy to configure a web service in a standard way without the service needing to know ahead of time the type of key or algorithm that will be used.  Today the service uses only an ECDSA key and the ES256 algorithm and to change that would require changes to code, but this config will allow us to specify any key and algorithm supported by the `gopkg.in/square/go-jose.v2` package.

This change is an accompanying change to #2450.

Related to #2443.

[RFC7517]: https://tools.ietf.org/html/rfc7517

### Known limitations

This change is a breaking change for the service, but since the service is in the `exp` experimental package we have the capability to make breaking changes.

### CODE REVIEW

**Comment by leighmcculloch** [comment]
### TODO

- [ ] Add a command for converting existing ASN.1 DER Base64 encoded private and public keys into JSON Web Keys.

File: exp/services/webauth/cmd/convjwtkey.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"gopkg.in/square/go-jose.v2"
+)
+
+type ConvJWTKeyCommand struct {
+	Logger *supportlog.Entry
+}
+
+func (c *ConvJWTKeyCommand) Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "convjwtkey [key]",
```
Comment (accordeiro) [inline exp/services/webauth/cmd/convjwtkey.go:18]: This is definitely not critical, but I'd rather be explicit in the command name about what we're converting to what. I feel like `jwt2jwk` might paint a clearer picture of what the command takes as an argument and what it outputs. Does that make sense?

File: exp/services/webauth/cmd/serve.go [hunk 2]
```diff
@@ -61,10 +61,10 @@ func (c *ServeCommand) Command() *cobra.Command {
 			Required:       true,
 		},
 		{
-			Name:      "jwt-key",
-			Usage:     "Base64 encoded ECDSA private key used for signing JWTs",
+			Name:      "jwk",
+			Usage:     "JSON Web Key (JWK) used for signing JWTs",
```
Comment (accordeiro) [inline exp/services/webauth/cmd/serve.go:65]: I understand there are symmetric keys that would render the private/public concept relevant. But maybe we could add a hint? Something like:

`If you have a public-private pair, this should be the private one.`

File: exp/services/webauth/cmd/convjwtkey.go (hunk 1 above)
Comment (leighmcculloch) [reply exp/services/webauth/cmd/convjwtkey.go]: Great idea, I love this feedback. In d526953d I've renamed this to convjwtkey2jwk, and I've already renamed genjwtkey to genjwk, so if you were to look at those commands they tell a story that you used to genjwtkey, and now you genjwk, and you can use convjwtkey2jwk to convert a key between that old and new format.

File: exp/services/webauth/cmd/serve.go (hunk 2 above)
Comment (leighmcculloch) [reply exp/services/webauth/cmd/serve.go]: 💯 8e40c60d

**Pull request merged**